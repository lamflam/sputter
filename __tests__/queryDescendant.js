const snapshot = require("../testUtil/snapshot");
const fixtures = require("../testUtil/fixtures");

describe("Descendant query", () => {
  describe("conditional matches", () => {
    snapshot(fixtures.conditional, "Program IfStatement");
  });

  describe("#8: descendant selector includes ancestor in search", () => {
    snapshot(fixtures.conditional, "Identifier[name=x]");
    snapshot(fixtures.conditional, "Identifier [name=x]");
    snapshot(fixtures.conditional, "BinaryExpression [name=x]");
    snapshot(fixtures.conditional, "AssignmentExpression [name=x]");
  });
});
