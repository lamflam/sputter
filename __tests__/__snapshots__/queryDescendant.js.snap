// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pseudo matches query #8: descendant selector includes ancestor in search AssignmentExpression [name=x] 1`] = `
  2 |   foo();
  3 | } else {
> 4 |   x = 2;
    |   ^ Identifier
  5 | }
  6 | if ((x == "test" && true) || x) {
  7 |   y = -1;
`;

exports[`Pseudo matches query #8: descendant selector includes ancestor in search BinaryExpression [name=x] 1`] = `
> 1 | if (x === 1) {
    |     ^ Identifier
  2 |   foo();
  3 | } else {
  4 |   x = 2;
  4 |   x = 2;
  5 | }
> 6 | if ((x == "test" && true) || x) {
    |      ^ Identifier
  7 |   y = -1;
  8 | } else if (false) {
  9 |   y = 1;
`;

exports[`Pseudo matches query #8: descendant selector includes ancestor in search Identifier [name=x] 1`] = ``;

exports[`Pseudo matches query #8: descendant selector includes ancestor in search Identifier[name=x] 1`] = `
> 1 | if (x === 1) {
    |     ^ Identifier
  2 |   foo();
  3 | } else {
  4 |   x = 2;
  2 |   foo();
  3 | } else {
> 4 |   x = 2;
    |   ^ Identifier
  5 | }
  6 | if ((x == "test" && true) || x) {
  7 |   y = -1;
  4 |   x = 2;
  5 | }
> 6 | if ((x == "test" && true) || x) {
    |      ^ Identifier
  7 |   y = -1;
  8 | } else if (false) {
  9 |   y = 1;
  4 |   x = 2;
  5 | }
> 6 | if ((x == "test" && true) || x) {
    |                              ^ Identifier
  7 |   y = -1;
  8 | } else if (false) {
  9 |   y = 1;
`;

exports[`Pseudo matches query conditional matches Program IfStatement 1`] = `
> 1 | if (x === 1) {
    | ^^^^^^^^^^^^^
> 2 |   foo();
    | ^^^^^^^^
> 3 | } else {
    | ^^^^^^^^
> 4 |   x = 2;
    | ^^^^^^^^
> 5 | }
    | ^^ IfStatement
  6 | if ((x == "test" && true) || x) {
  7 |   y = -1;
  8 | } else if (false) {
   4 |   x = 2;
   5 | }
>  6 | if ((x == "test" && true) || x) {
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>  7 |   y = -1;
     | ^^^^^^^^^
>  8 | } else if (false) {
     | ^^^^^^^^^
>  9 |   y = 1;
     | ^^^^^^^^^
> 10 | }
     | ^^ IfStatement
  11 | 
   6 | if ((x == "test" && true) || x) {
   7 |   y = -1;
>  8 | } else if (false) {
     |        ^^^^^^^^^^^
>  9 |   y = 1;
     | ^^^^^^^^
> 10 | }
     | ^^ IfStatement
  11 | 
`;
